File: week1/Exercise_1_vb/Exercise_1_vb/Form1.vb
Imports System.Text

Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        DataGridView1.Rows.Clear()
        OpenFileDialog1.Filter = "CSV files (*.csv)|*.CSV"
        If OpenFileDialog1.ShowDialog <> Windows.Forms.DialogResult.Cancel Then
            TextBox1.Text = OpenFileDialog1.FileName
        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        ReadCSV(TextBox1.Text)
    End Sub

    Private Sub ReadCSV(fname As String)
        Dim TextLine As String = ""
        Dim SplitLine() As String

        Using objReader As New System.IO.StreamReader(fname, Encoding.ASCII)
            Do While objReader.Peek() <> -1
                TextLine = objReader.ReadLine()
                SplitLine = Split(TextLine, ",")
                Me.DataGridView1.Rows.Add(SplitLine)
            Loop
        End Using
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        RichTextBox1.Text = "Lowest grade in the class" & vbCrLf & "Grade: " + (From row As DataGridViewRow In DataGridView1.Rows
                                                                                Where row.Cells(1).FormattedValue.ToString() <> String.Empty
                                                                                Select Convert.ToInt32(row.Cells(1).FormattedValue)
                             ).Min().ToString()
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        RichTextBox1.Text = "Highest grade in the class" & vbCrLf & "Grade: " + (From row As DataGridViewRow In DataGridView1.Rows
                                                                                 Where row.Cells(1).FormattedValue.ToString() <> String.Empty
                                                                                 Select Convert.ToInt32(row.Cells(1).FormattedValue)
                             ).Max().ToString()
    End Sub

    Private Sub Button5_Click(sender As Object, e As EventArgs) Handles Button5.Click
        RichTextBox1.Text = "Arithmetic mean of the class" & vbCrLf & "Average: " + (From row As DataGridViewRow In DataGridView1.Rows
                                                                                     Where row.Cells(1).FormattedValue.ToString() <> String.Empty
                                                                                     Select Convert.ToInt32(row.Cells(1).FormattedValue)
                             ).Average().ToString("00.00")
    End Sub
End Class

File: week1/Exercise_1/Exercise_1/Form1.cs
using System.DirectoryServices.ActiveDirectory;

namespace Exercise_1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            OpenFileDialog dlg = new OpenFileDialog();
            dlg.Filter = "Files|*.csv";
            dlg.ShowDialog();
            textBox1.Text = dlg.FileName;
        }

        public List<Student> LoadCSV(string csvFile)
        {
            var query = from l in File.ReadAllLines(csvFile)
                        let data = l.Split(",")
                        select new Student
                        {
                            Name = data[0],
                            Grade = int.Parse(data[1])
                        };
            return query.ToList();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            dataGridView1.DataSource = LoadCSV(textBox1.Text);
        }

        private void button5_Click(object sender, EventArgs e)
        {
            string lowest = (from DataGridViewRow row in dataGridView1.Rows
                            where row.Cells[1].FormattedValue.ToString() != String.Empty
                            select Convert.ToInt32(row.Cells[1].FormattedValue)).Min().ToString();

            richTextBox1.Text = "Lowest grade in the class\nGrade: " + lowest;
        }

        private void button4_Click(object sender, EventArgs e)
        {
            string highest = (from DataGridViewRow row in dataGridView1.Rows
                              where row.Cells[1].FormattedValue.ToString() != String.Empty
                              select Convert.ToInt32(row.Cells[1].FormattedValue)).Max().ToString();
            richTextBox1.Text = "Highest grade in the class\nGrade: " + highest;
        }

        private void button3_Click(object sender, EventArgs e)
        {
            string a_mean = (from DataGridViewRow row in dataGridView1.Rows
                      where row.Cells[1].FormattedValue.ToString() != String.Empty
                      select Convert.ToInt32(row.Cells[1].FormattedValue)).Average().ToString("00.00");
            richTextBox1.Text = "Arithmetic mean of the class\nAverage: " + a_mean;

        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }

    public class Student
    {
        public string Name { get; set; }
        public int Grade { get; set; }

    }
}